README.md
<h3 align="center">
  <a href="https://github.com/Rain120/busuanzi-statistics">Bu Suan Zi Statistics</a>
  <img src='./statistics.png' alt='statistics' width='100%' height='40%' />
</h3>

<div align="center">

[![GitHub watchers](https://img.shields.io/github/watchers/rain120/busuanzi-statistics?style=social)](https://github.com/Rain120/busuanzi-statistics/watchers)
[![STAR](https://img.shields.io/github/stars/rain120/busuanzi-statistics?style=social)](https://github.com/Rain120/busuanzi-statistics/stargazers) [![FORK](https://img.shields.io/github/forks/rain120/busuanzi-statistics?style=social)](https://github.com/Rain120/busuanzi-statistics/network/members)

[![npm download](https://img.shields.io/npm/dt/busuanzi-statistics?style=flat-square)](https://www.npmjs.com/package/busuanzi-statistics) [![jsDelivr hits (npm)](https://img.shields.io/jsdelivr/npm/hy/busuanzi-statistics?style=flat-square)](https://www.jsdelivr.com/package/npm/busuanzi-statistics) [![jsdelivr](https://data.jsdelivr.com/v1/package/npm/busuanzi-statistics/badge)](https://www.jsdelivr.com/package/npm/busuanzi-statistics)
[![ISSUES](https://img.shields.io/github/issues/rain120/busuanzi-statistics?style=flat-square)](https://github.com/Rain120/busuanzi-statistics/issues) [![GitHub closed pull requests](https://img.shields.io/github/issues-pr-closed/rain120/busuanzi-statistics?style=flat-square)](https://github.com/Rain120/busuanzi-statistics/pulls) [![COMMIT](https://img.shields.io/github/last-commit/rain120/busuanzi-statistics?style=flat-square)](https://github.com/Rain120/busuanzi-statistics/commits/master) ![LANGUAGES](https://img.shields.io/github/languages/top/rain120/busuanzi-statistics?style=flat-square)
[![npm](https://img.shields.io/npm/v/busuanzi-statistics?style=flat-square)](https://www.npmjs.com/package/busuanzi-statistics)
<!-- [![VERSION](https://img.shields.io/github/package-json/v/rain120/busuanzi-statistics?style=flat-square)](https://github.com/Rain120/busuanzi-statistics/blob/master/package.json) -->

</div>

## 😚 Welcome

Welcome to the Bu Suan Zi Statistics.

不蒜子访问量统计功能

#### 🎮 TL;DR

⌨️ How to use it, 👇

#### 🔨 Usage

##### Node

```md
npm install busuanzi-statistics --save

// or

yarn add busuanzi-statistics
```

```js
import statistics from 'busuanzi-statistics';

document.querySelector('body').addEventListener('onloaded', () => {
    statistics.fetch();
}, false);

```

##### CDN

```md
<script src="https://cdn.jsdelivr.net/npm/busuanzi-statistics@0.0.2/lib/busuanzi-statistics.min.js"></script>
```

#### 🤝 Contributing ![PR](https://img.shields.io/badge/PRs-Welcome-orange?style=flat-square&logo=appveyor)

We welcome all contributions. You can submit any ideas as [pull requests](https://github.com/Rain120/busuanzi-statistics/pulls) or as a GitHub [issue](https://github.com/Rain120/busuanzi-statistics/issues).

#### 🔗 Links

[NPM Package](https://www.npmjs.com/package/busuanzi-statistics)

[CDN Package](https://www.jsdelivr.com/package/npm/busuanzi-statistics)

[Bu Suan Zi](https://busuanzi.ibruce.info/)

#### 👨‍🏭 Author

> Front-End development engineer, technology stack: React + Typescript + Mobx, also used Vue + Vuex for a while

- [Github](https://github.com/Rain120)
- [知乎](https://www.zhihu.com/people/yan-yang-nian-hua-120/activities)
- [掘金](https://juejin.im/user/57c616496be3ff00584f54db)

#### 📝 License

[![LICENSE](https://img.shields.io/github/license/rain120/busuanzi-statistics?style=flat-square)](https://github.com/Rain120/busuanzi-statistics/blob/master/LICENSE)

Copyright © 2020-present [Rain120](https://github.com/Rain120).



CHANGELOG.md
<a name="0.0.2"></a>
## 0.0.2 (2020-02-15)


### Features

* busuanzi statistics ([780d7e9](https://github.com/Rain120/busuanzi-statistics/commit/780d7e9))
* update package main; README badge, CDN, Link ([adda5f8](https://github.com/Rain120/busuanzi-statistics/commit/adda5f8))





lib/busuanzi-statistics.js
// busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js

var bszCaller, bszTag, scriptTag, ready;

var t,
    e,
    n,
    a = false,
    c = [];

if (typeof document !== "undefined") {
  ready = function (t) {
    return a || "interactive" === document.readyState || "complete" === document.readyState ? t.call(document) : c.push(function () {
      return t.call(this);
    }), this;
  }, e = function () {
    for (var t = 0, e = c.length; t < e; t++) c[t].apply(document);
    c = [];
  }, n = function () {
    a || (a = !0, e.call(window), document.removeEventListener ? document.removeEventListener("DOMContentLoaded", n, false) : document.attachEvent && (document.detachEvent("onreadystatechange", n), window == window.top && (clearInterval(t), t = null)));
  }, document.addEventListener ? document.addEventListener("DOMContentLoaded", n, false) : document.attachEvent && (document.attachEvent("onreadystatechange", function () {
    /loaded|complete/.test(document.readyState) && n();
  }), window == window.top && (t = setInterval(function () {
    try {
      a || document.documentElement.doScroll("left");
    } catch (t) {
      return;
    }
    n();
  }, 5)));
}

bszCaller = {
  fetch: function (t, e) {
    var n = "BusuanziCallback_" + Math.floor(1099511627776 * Math.random());
    t = t.replace("=BusuanziCallback", "=" + n);
    scriptTag = document.createElement("SCRIPT"), scriptTag.type = "text/javascript", scriptTag.defer = !0, scriptTag.src = t, document.getElementsByTagName("HEAD")[0].appendChild(scriptTag);
    window[n] = this.evalCall(e);
  },
  evalCall: function (e) {
    return function (t) {
      ready(function () {
        try {
          e(t), scriptTag && scriptTag.parentElement && scriptTag.parentElement.removeChild && scriptTag.parentElement.removeChild(scriptTag);
        } catch (t) {
          console.log(t), bszTag.hides();
        }
      });
    };
  }
};

bszTag = {
  bszs: ["site_pv", "page_pv", "site_uv"],
  texts: function (n) {
    this.bszs.map(function (t) {
      var e = document.getElementById("busuanzi_value_" + t);
      e && (e.innerHTML = n[t]);
    });
  },
  hides: function () {
    this.bszs.map(function (t) {
      var e = document.getElementById("busuanzi_container_" + t);
      e && (e.style.display = "none");
    });
  },
  shows: function () {
    this.bszs.map(function (t) {
      var e = document.getElementById("busuanzi_container_" + t);
      e && (e.style.display = "inline");
    });
  }
};

const fetch = () => {
  bszTag && bszTag.hides();
  bszCaller.fetch("//busuanzi.ibruce.info/busuanzi?jsonpCallback=BusuanziCallback", function (t) {
    bszTag.texts(t), bszTag.shows();
  });
};

if (typeof document !== "undefined") {
  fetch();
}

if (typeof module !== "undefined" && typeof module.exports !== "undefined") {
  module.exports = fetch;
} else {
  if (typeof define === "function" && define.amd) {
    define([], function () {
      return fetch;
    });
  } else {
    window.fetch = fetch;
  }
}


src/index.js
// busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js

var bszCaller, bszTag, scriptTag, ready;

var t,
  e,
  n,
  a = false,
  c = [];

if (typeof document !== "undefined") {
  (ready = function(t) {
    return (
      a ||
      "interactive" === document.readyState ||
      "complete" === document.readyState
        ? t.call(document)
        : c.push(function() {
            return t.call(this);
          }),
      this
    );
  }),
    (e = function() {
      for (var t = 0, e = c.length; t < e; t++) c[t].apply(document);
      c = [];
    }),
    (n = function() {
      a ||
        ((a = !0),
        e.call(window),
        document.removeEventListener
          ? document.removeEventListener("DOMContentLoaded", n, false)
          : document.attachEvent &&
            (document.detachEvent("onreadystatechange", n),
            window == window.top && (clearInterval(t), (t = null))));
    }),
    document.addEventListener
      ? document.addEventListener("DOMContentLoaded", n, false)
      : document.attachEvent &&
        (document.attachEvent("onreadystatechange", function() {
          /loaded|complete/.test(document.readyState) && n();
        }),
        window == window.top &&
          (t = setInterval(function() {
            try {
              a || document.documentElement.doScroll("left");
            } catch (t) {
              return;
            }
            n();
          }, 5)));
}

bszCaller = {
  fetch: function(t, e) {
    var n = "BusuanziCallback_" + Math.floor(1099511627776 * Math.random());
    t = t.replace("=BusuanziCallback", "=" + n);
    (scriptTag = document.createElement("SCRIPT")),
      (scriptTag.type = "text/javascript"),
      (scriptTag.defer = !0),
      (scriptTag.src = t),
      document.getElementsByTagName("HEAD")[0].appendChild(scriptTag);
    window[n] = this.evalCall(e);
  },
  evalCall: function(e) {
    return function(t) {
      ready(function() {
        try {
          e(t),
            scriptTag &&
              scriptTag.parentElement &&
              scriptTag.parentElement.removeChild &&
              scriptTag.parentElement.removeChild(scriptTag);
        } catch (t) {
          console.log(t), bszTag.hides();
        }
      });
    };
  }
};

bszTag = {
  bszs: ["site_pv", "page_pv", "site_uv"],
  texts: function(n) {
    this.bszs.map(function(t) {
      var e = document.getElementById("busuanzi_value_" + t);
      e && (e.innerHTML = n[t]);
    });
  },
  hides: function() {
    this.bszs.map(function(t) {
      var e = document.getElementById("busuanzi_container_" + t);
      e && (e.style.display = "none");
    });
  },
  shows: function() {
    this.bszs.map(function(t) {
      var e = document.getElementById("busuanzi_container_" + t);
      e && (e.style.display = "inline");
    });
  }
};

const fetch = () => {
  bszTag && bszTag.hides();
  bszCaller.fetch(
    "//busuanzi.ibruce.info/busuanzi?jsonpCallback=BusuanziCallback",
    function(t) {
      bszTag.texts(t), bszTag.shows();
    }
  );
};

if (typeof document !== "undefined") {
  fetch();
}

if (typeof module !== "undefined" && typeof module.exports !== "undefined") {
  module.exports = fetch;
} else {
  if (typeof define === "function" && define.amd) {
    define([], function() {
      return fetch;
    });
  } else {
    window.fetch = fetch;
  }
}
